src/DynamoPublish/Models/PublishModel.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Dynamo.Core;
+using Dynamo.Interfaces;
+using Dynamo.Models;
+using Dynamo.Wpf.Authentication;
+using Greg.AuthProviders;
+using Reach;
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Dynamo.Publish.Models
+{
+    public class PublishModel
+    {
+        private readonly AuthenticationManager manager;
+        public LoginService LoginService;
+
+        private readonly string serverUrl;
+        private readonly string port;
+        private readonly string page;
+        private readonly string provider;
+
+        public bool IsLoggedIn
+        {
+            get
+            {
+                return manager.LoginState == LoginState.LoggedIn;
+            }
+        }
+
+        #region Initialization
+
+        public PublishModel()
+        {
+            // Open the configuration file using the dll location.
+            Configuration config = ConfigurationManager.OpenExeConfiguration(this.GetType().Assembly.Location);
+            // Get the appSettings section.
+            AppSettingsSection appSettings = (AppSettingsSection)config.GetSection("appSettings");
+
+            serverUrl = appSettings.Settings["ServerUrl"].Value;
+            port = appSettings.Settings["Port"].Value;
+            page = appSettings.Settings["Page"].Value;
+            provider = appSettings.Settings["Provider"].Value;
+
+            if (!String.IsNullOrWhiteSpace(provider))
+                manager = new AuthenticationManager(new OxygenProvider(provider));
+        }
+
+        #endregion
+
+        internal void Authenticate()
+        {
+            if (manager == null)
+                throw new Exception(Resource.AuthenticationErrorMessage);
+
+            if (manager.HasAuthProvider && LoginService != null)
+            {
+                manager.AuthProvider.RequestLogin += LoginService.ShowLogin;
+                manager.AuthProvider.Login();
+            }
+        }
+
+        /// <summary>
+        /// Sends workspace and its' dependencies to Flood.
+        /// </summary>
+        internal void SendWorkspaces(IEnumerable<IWorkspaceModel> workspaces)
+        {
+            if (String.IsNullOrWhiteSpace(serverUrl) || String.IsNullOrWhiteSpace(manager.Username))
+                return;
+
+            string fullServerAdress = serverUrl + ":" + port;
+            var reachClient = new WorkspaceStorageClient(manager.AuthProvider, fullServerAdress);
+            var result = reachClient.Send(workspaces.OfType<HomeWorkspaceModel>().First(), workspaces.OfType<CustomNodeWorkspaceModel>());
```

Comment:
This code is unfortunately not correct.  IOW, the `CustomNodeWorkspaceModel`'s are not the dependencies.  

In order to find the dependencies of a `Workspace`, we need to:

1.  find the `Function` nodes and their respective dependencies
2.  get the union of all of them.  

Getting the `Dependencies` field of a `Definition` will allow you to do that.